packages/gestalt/src/DropdownLink.js

```diff
@@ -1,31 +1,55 @@
 // @flow strict
 import { type Node } from 'react';
-import OptionItem, { type OptionItemType } from './OptionItem.js';
+import OptionItem from './OptionItem.js';
 import { DropdownContextConsumer } from './DropdownContext.js';
 import { type AbstractEventHandler } from './AbstractEventHandler.js';
 
-type PublicProps = {|
+type OptionItemType = {|
+  label: string,
+  subtext?: string,
+  value: string,
+|};
+
+type Props = {|
+  /**
+   * When supplied, will display a [Badge](https://gestalt.pinterest.systems/badge) next to the item's label. See the [Badges](https://gestalt.pinterest.systems/dropdown#Badges) variant to learn more.
+   */
   badgeText?: string,
+  /**
+   * If needed, users can supply custom content to each Dropdown Link. This can be useful when extra functionality is needed beyond a basic Link. See the [Custom item content](https://gestalt.pinterest.systems/dropdown#Custom-item-content) variant to learn more.
+   */
   children?: Node,
+  /**
+   * When supplied, will add a data-test-id prop to the dom element.
+   */
   dataTestId?: string,
+  /**
+   * Directs users to the url when item is selected. See the [Types of items](https://gestalt.pinterest.systems/dropdown#Types-of-items) variant to learn more.
+   */
   href: string,
+  /**
+   * When true, adds an arrow icon to the end of the item to signal this item takes users to an external source and opens the link in a new tab. Do not add if the item navigates users within the app. See the [Best practices](https://gestalt.pinterest.systems/dropdown#Best-practices) for more info.
+   */
   isExternal?: boolean,
+  /**
+   * Callback fired when clicked (pressed and released) with a mouse or keyboard. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/onlinknavigationprovider) to learn more about link navigation.
+   */
   onClick?: AbstractEventHandler<
```

Comment:
I've been manually copying over the type of AbstractEventHandler, since that abstraction means nothing to the docs reader. It's a pretty easy change, and IMHO makes the type a bit more readable as well.

Reply:
DONE